### PR TITLE
fix: add domain set primary bool

### PR DESCRIPTION
**Pull request by stebenz**

### CODE REVIEW

File: zitadel/v2/domain/resource.go [hunk 1]
```diff
@@ -27,7 +27,7 @@ func GetResource() *schema.Resource {
 			},
 			isPrimaryVar: {
 				Type:        schema.TypeBool,
-				Computed:    true,
+				Optional:    true,
```
Comment (eliobischof) [inline zitadel/v2/domain/resource.go:30]: So just for my understanding:
Terraform replaces the resource automatically, aka deletes and creates it, if we don't specify an update callback? Right?

**Review by eliobischof — CHANGES_REQUESTED** [review]

File: zitadel/v2/domain/funcs.go [hunk 2]
```diff
@@ -56,6 +56,10 @@ func create(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Dia
 		return diag.Errorf("failed to create domain: %v", err)
 	}
 	d.SetId(name)
+	_, err = client.SetPrimaryOrgDomain(ctx, &management.SetPrimaryOrgDomainRequest{Domain: name})
```
[the remark this answers is not in the document]
Comment (stebenz) [reply zitadel/v2/domain/funcs.go:59]: you're right, checked with an if

File: zitadel/v2/domain/resource.go (hunk 1 above)
Comment (stebenz) [reply zitadel/v2/domain/resource.go:30]: you're right, added an update function which only triggers when isprimary is changed and sets primary if true

Comment (eliobischof) [reply zitadel/v2/domain/resource.go:30]: But that disables changing a domain name, right? Replacing is probably better?

**Comment by eliobischof** [comment]
I unresolved a comment @stebenz

File: zitadel/v2/domain/resource.go (hunk 1 above)
Comment (stebenz) [reply zitadel/v2/domain/resource.go:30]: there is no changing a domain name as the domain name is the primary key, and I created the update function as the primary attribute is separate and if you change the boolean you have to call ane endpoint

Comment (eliobischof) [reply zitadel/v2/domain/resource.go:30]: in a terraform resources perspective, we could replace a domain you implemented initially, effectively enabling changing a domain name of a specific terraform resource.
But fine by me if we decide against it.

**Review by eliobischof — APPROVED** [review]

**Pull request merged**